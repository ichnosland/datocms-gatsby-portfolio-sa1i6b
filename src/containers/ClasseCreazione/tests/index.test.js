import React from 'react';
import { shallow } from 'enzyme';

import { ClasseCreazioneView } from '../index';

const mockConfiguration = {
  product: 'product',
  hasPremium: true,
  disciplinaId: 123,
};

const mockProps = {
  configuration: mockConfiguration,
  onResetForm: () => { },
  onClasseCreazioneReset: () => { },
  onClasseCreazioneScuoleAttiveFetch: () => { },
  onClasseCreazioneDataSet: () => { },
  onClasseCreazioneDataPost: () => { },
  onClasseCreazioneGeoFetch: () => { },
  onClasseCreazioneGeoComuneFetch: () => { },
  onClasseCreazioneScuoleAttiveSet: () => { },
  spinner: false,
  feedback: {
    hasFeedback: false,
    tipologia: '',
    errorMessage: '',
  },
  history: {
    push: () => { },
  },
  scuoleAttive: {
    list: [],
    isLoaded: true,
  },
  geo: {
    province: [],
    isProvinceLoaded: false,
    comuni: {},
    scuole: {},
    indirizziDiStudio: [],
  },
  data: {
    display: '',
    creaNuovaClasse: false,
    scuolaSelezionata: {
      pk: 2,
      nome: 'nome',
      classi: [],
    },
    nuova: false,
  },
};

describe('<RegistrationView />', () => {
  it('should match snapshot', () => {
    const renderedComponent = shallow(
      <ClasseCreazioneView {...mockProps} />
    );

    expect(renderedComponent).toMatchSnapshot();
  });
});
