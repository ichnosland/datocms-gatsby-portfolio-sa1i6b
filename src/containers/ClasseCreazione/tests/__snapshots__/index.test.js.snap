// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationView /> should match snapshot 1`] = `
<div>
  <FormElements__FormSection
    paddingmobile="20px"
  >
    <ReduxForm
      data={
        Object {
          "cercaAltraScuola": Object {
            "comuni": Array [],
            "indirizziDiStudio": Array [],
            "isComuneSelezionato": false,
            "isIndirizzoDiStudioSelezionato": false,
            "isProvinciaSelezionata": false,
            "isScuolaSelezionata": true,
            "province": Array [],
            "scuole": Array [],
            "selezionaComune": [Function],
            "selezionaIndirizzoDiStudio": [Function],
            "selezionaProvincia": [Function],
            "selezionaScuola": [Function],
          },
          "classeDocente": Object {
            "aggiungiCampoNuovaClasse": [Function],
            "creaDaClasseEsistenteFunction": [Function],
            "creaNuovaClasseFunction": [Function],
            "isClasseSelezionata": false,
            "isCreaNuovaClasse": false,
            "isDatiNuovaClasseValid": undefined,
            "items": Array [],
            "onChangeDatiClasse": [Function],
            "selectionFx": [Function],
            "titolo": "nome",
          },
          "scuoleAttive": Object {
            "cercaAltraScuolaFunction": [Function],
            "items": Array [],
            "selectionFx": [Function],
          },
        }
      }
      display=""
      product="product"
    />
    <ButtonGroup
      buttons={
        Array [
          Object {
            "actioncolor": "action",
            "id": 1,
            "label": "Cancella",
            "onClickFunction": [Function],
          },
          Object {
            "id": 2,
            "label": "Esci",
            "onClickFunction": [Function],
          },
        ]
      }
    />
  </FormElements__FormSection>
</div>
`;
